route/build/src/dataTypes.f90

```diff
@@ -158,32 +148,36 @@ MODULE dataTypes
-   integer(i4b)             , allocatable  :: hru_id(:)     ! id of HM_HRUs or RN_HRUs at which runoff is stored (size: nSpace(1))
-   integer(i4b)             , allocatable  :: hru_ix(:)     ! Index of RN_HRUs associated with river network (used only if HM_HRUs = RN_HRUs)
-   real(dp)                 , allocatable  :: basinRunoff(:)! remapped river network catchment runoff (size: number of nHRU)
-   real(dp)                 , allocatable  :: basinEvapo(:) ! remapped river network catchment runoff (size: number of nHRU)
-   real(dp)                 , allocatable  :: basinPrecip(:)! remapped river network catchment runoff (size: number of nHRU)
+ ! mapping time step between two time series e.g., simulation time step vs runoff time step for one simulation time step
+ type, public :: map_time
+   integer(i4b), allocatable :: iFile(:)
+   integer(i4b), allocatable :: iTime(:)
+   real(dp),     allocatable :: frac(:)
+ end type map_time
+
+ ! ---------- forcing and water management data  ----------------------------------------------------------------------
+
+ type, public :: inputData
+   integer(i4b)                            :: nSpace(1:2)     ! number of spatial dimension
+   real(dp)                 , allocatable  :: sim(:)          ! flux simulation (HM_HRU) at one time step (size: nSpace(1))
+   real(dp)                 , allocatable  :: sim2d(:,:)      ! flux simulation (x,y) at one time step (size: /nSpace(1),nSpace(2)/)
+   real(dp)                                :: fillvalue       ! fillvalue
+ end type inputData
+
+ type, public, extends(inputData) :: runoff
+   integer(i4b)             , allocatable  :: hru_id(:)       ! id of HM_HRUs or RN_HRUs at which runoff is stored (size: nSpace(1))
+   integer(i4b)             , allocatable  :: hru_ix(:)       ! Index of RN_HRUs associated with river network (used only if HM_HRUs = RN_HRUs)
+   real(dp)                 , allocatable  :: basinRunoff(:)  ! remapped river network catchment runoff (size: number of nHRU)
+   real(dp)                 , allocatable  :: basinEvapo(:)   ! remapped river network catchment runoff (size: number of nHRU)
+   real(dp)                 , allocatable  :: basinPrecip(:)  ! remapped river network catchment runoff (size: number of nHRU)
  end type runoff
 
- ! water management data; fluxes to/from reaches or target volume
- type, public :: wm
-   integer(i4b)                            :: nTime           ! number of time steps
-   integer(i4b)                            :: nSpace(1:2)     ! number of spatial dimension, in this case only one dimentonal
-   real(dp)                                :: time            ! time variable at one time step
-   real(dp)                 , allocatable  :: sim(:)          ! user specified flux add/subtract, or volume at one time step (size: nSpace)
-   real(dp)                 , allocatable  :: sim2D(:,:)      ! to provide modularity for reading data
+ type, public, extends(inputData) :: wm
```

Comment:
We should add some comments that wm means Water Management

Reply:
Added a comment "water-management" next to `wm`